cmd/lotus-shed/fip-0036.go

```diff
@@ -0,0 +1,565 @@
+								msigPendingVotes[ms] = mpv
+							}
+						}
+					} else { //first vote received from one of the signers of the msig
+						msi, ok := msigActorsInfo[ms]
+						if !ok {
+							return xerrors.Errorf("didn't find msig %s in msig map", ms)
+						}
+
+						if msi.Threshold == 1 { //met threshold with this signer's single vote
+							if v == APPROVE {
+								approveBalance = types.BigAdd(approveBalance, msi.Balance)
+								votesIncludingMsigs[ms] = APPROVE
+
+							} else {
+								rejectionBalance = types.BigAdd(rejectionBalance, msi.Balance)
+								votesIncludingMsigs[ms] = Reject
+							}
+							votedMsigs[ms] = struct{}{}
+						} else { //threshold not met, add to pending vote
+							if v == APPROVE {
+								msigPendingVotes[ms] = msigVote{
+									Multisig:     msi,
+									ApproveCount: 1,
+								}
+							} else {
+								msigPendingVotes[ms] = msigVote{
+									Multisig:    msi,
+									RejectCount: 1,
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+
+		// time to process miners based on what we know about votesIncludingMsigs
+		minerVotes := make(map[address.Address]uint64)
+		fmt.Println("counting miner votes")
+		for s, v := range votes {
+			if minerInfos, found := ownerMap[s]; found {
+				for _, minerInfo := range minerInfos {
+					minerVotes[minerInfo] = v
+				}
+			}
+			if minerInfos, found := workerMap[s]; found {
+				for _, minerInfo := range minerInfos {
+					if _, ok := minerVotes[minerInfo]; !ok {
+						minerVotes[minerInfo] = v
+					}
+				}
+			}
+		}
+
+		approveRBP := big.Zero()
+		approveDealPower := big.Zero()
+		rejectionRBP := big.Zero()
+		rejectionDealPower := big.Zero()
+		fmt.Println("adding up miner votes")
+		for minerAddr, vote := range minerVotes {
+			mbi, ok := minerActorsInfo[minerAddr]
+			if !ok {
+				return xerrors.Errorf("failed to find miner info for %s", minerAddr)
+			}
+
+			if vote == APPROVE {
+				approveBalance = big.Add(approveBalance, mbi.balance)
+				approveRBP = big.Add(approveRBP, mbi.rawBytePower)
+				approveDealPower = big.Add(approveDealPower, mbi.dealPower)
+			} else {
+				rejectionBalance = big.Add(rejectionBalance, mbi.balance)
+				rejectionRBP = big.Add(rejectionRBP, mbi.rawBytePower)
+				rejectionDealPower = big.Add(rejectionDealPower, mbi.dealPower)
+			}
+		}
+
+		fmt.Println("Total acceptance token: ", approveBalance)
+		fmt.Println("Total rejection token: ", rejectionBalance)
+
+		fmt.Println("Total acceptance SP deal power: ", approveDealPower)
+		fmt.Println("Total rejection SP deal power: ", rejectionDealPower)
+
+		fmt.Println("Total acceptance SP rb power: ", approveRBP)
+		fmt.Println("Total rejection SP rb power: ", rejectionRBP)
+
+		fmt.Println("Total acceptance Client rb power: ", clientApproveBytes)
+		fmt.Println("Total rejection Client rb power: ", clientRejectBytes)
+
+		fmt.Println("\n\nFinal results **drumroll**")
+		if rejectionBalance.GreaterThanEqual(big.Mul(approveBalance, big.NewInt(2))) {
```

Comment:
According to https://github.com/filecoin-project/FIPs/discussions/464 veto is meant to be >75%, but here we'd say veto if 66% reject

Reply:
Yah can we simply use reject / (reject + approve) >.75, dear master mathematician?

Reply:
@magik6k Are you saying I can't type 2 and expect it to be 3?

Reply:
Thanks for the catch!